[IMP] point_of_sale: add Cashmatic cash machine docs (#18242)
odoo/documentation: content/applications/sales/point_of_sale/payment_methods/cash_machines/cashmatic.rst

- On the Cashmatic machine, log in with the provided credentials, then go to
  :menuselection:`Settings --> IP address`.
- In the MyCashmatic app or on the website, sign in to your account, then go to
  :menuselection:`Settings --> Network`.

Note the IP address displayed on the machine's or the device's screen, as it is needed for the
:ref:`Odoo configuration step <pos/cashmatic/odoo_configuration>`.

.. _pos/cashmatic/https:

Connection over HTTPS
---------------------

To connect the Cashmatic machine over HTTPS, a valid certificate is required. To generate and
download the certificate, follow these steps:

#. Open a web browser and type the IP address in the address bar to access the Cashmatic web
   interface.
#. In the interface's side menu, go to :menuselection:`Settings --> Device`.
#. Scroll down to the :guilabel:`Use Cashmatic CA` option and enable it.
#. Click :guilabel:`Save` and refresh the page. A :guilabel:`Download HTTPS Certificate` link
   appears at the bottom left of the side menu.
#. Click the :guilabel:`Download HTTPS Certificate` link.

To install the certificate on the device running Odoo Point of Sale, contact your integration

> partner, as the process depends on the computer's operating system.

Reviewer: ```suggestion
partner, as the process depends on the device's operating system.
```

Change 'computer' to 'device'?

=========
Cashmatic
=========

> `Cashmatic <https://cashmatic.eu/>`_ :doc:`cash machines <../cash_machines>` automate cash

Reviewer: I notice we don't link to the provider on the Glory or Cashdro pages ... if you think it's worth having, might as well harmonise across all three pages?

transactions by accepting cash and dispensing change. When integrated with Odoo Point of Sale,
payments are processed directly through the machine from the :ref:`POS register
<pos/use/open-register>`.

.. note::
   - The integration with the Cashmatic cash machine only supports payments and refunds.
   - Tasks such as filling and emptying the machine must be performed directly through the cash

>      machine interface or via the MyCasmatic app/website, available on `Android

Reviewer: ```suggestion
     machine interface or via the MyCashmatic app/website, available on `Android
```

     <https://play.google.com/store/apps/details?id=com.cashmatic.app&pcampaignid=web_share>`_ or

>      `iOS <https://apps.apple.com/us/app/mycashmatic/id1661294811>`_.

Reviewer: Couldn't we combine these into a single note?

Reviewer 2: I feel like these points cover two distinct topics, right?

Reviewer: Fine to keep it like it is :)

>      `iOS <https://apps.apple.com/us/app/mycashmatic/id1661294811>`_.

Reviewer: ```suggestion
     machine interface or via the MyCasmatic app, available on `Android
     <https://play.google.com/store/apps/details?id=com.cashmatic.app&pcampaignid=web_share>`_ or
     `iOS <https://apps.apple.com/us/app/mycashmatic/id1661294811>`_ or `website <https://cashmatic.be/my-cashmatic/login>`_.
```

Maybe we can just get the links to both the app and website out of the way in this first note?

.. _pos/cashmatic/configuration:

Cash machine configuration
==========================

.. important::
   - Two separate sets of credentials are set up during installation: one for the cash machine,
     required to :ref:`configure it in Odoo <pos/cashmatic/odoo_configuration>`, and one for the
     **MyCashmatic** app or `website <https://cashmatic.be/my-cashmatic/login>`_, required to

>      configure the cash machine itself.

Reviewer: I don't really follow here. When are these credentials 'set up'? (not during anything we explain below, right? do they just receive these from Cashmatic?) And maybe 'installation' > 'configuration'? Also, since we're covering maching config first, change the order of the mentioned sets?

Reviewer 2: Upon delivery of the machine, the credentials are provided (per email, I suppose). Should I use *provided* instead of *set up*?

Reviewer: 'Provided' seems good to me!

   - The Cashmatic machine must be assigned a static IP address to ensure stable operation.
   - The Cashmatic machine must be on the same local network as the device running Odoo Point of

>      Sale.

Reviewer: To reduce the number of bullets and avoid that repetition at the start of the sentences, could we combine these two related points? Also, if we do that, would it make sense to change the order, i.e., 

```suggestion
   - The Cashmatic machine must be on the same local network as the device running Odoo Point of
     Sale and must be assigned a static IP address to ensure stable operation.
```

   - Setting up the Cashmatic machine requires the hardware administrator's password and technical
     knowledge. Contact your Cashmatic integration partner for configuration support if assistance
     is needed.

To configure the Cashmatic machine as a :ref:`payment method in Odoo
<pos/cashmatic/odoo_configuration>`, first locate the machine's :ref:`IP address
<pos/cashmatic/ip-address>`. The IP address is required regardless of whether the connection uses
:ref:`HTTPS <pos/cashmatic/https>` or :ref:`HTTP <pos/cashmatic/http>`. While HTTPS provides a more

> secure connection, it requires additional configuration, whereas HTTP is simpler to set up.

Reviewer: ```suggestion
Configuring the Cashmatic machine as a :ref:`payment method in Odoo <pos/cashmatic/odoo_configuration>` requires locating the machine's :ref:`IP address <pos/cashmatic/ip-address>` and preparing for connection over :ref:`HTTPS <pos/cashmatic/https>` or :ref:`HTTP <pos/cashmatic/http>`. 

.. note::
   HTTPS provides a more secure connection and requires additional configuration, including installing a valid :abbr: `SSL (Secure Sockets Layer)` certificate. HTTP allows communication through :doc:`Local Network Access (LNA) <../../hardware_network/pos_lna>` and is recommended as it is simpler to set up.
```

- I wonder if it's better to keep the info about HTTPS vs HTTP to a self-contained note, (and also include the recommendation that's currently in a note in the Odoo configuration section).
- we mention 'SSL' in that note, so add it here too?

.. _pos/cashmatic/ip-address:

Locate the IP address
---------------------

> To locate the Cashmatic IP address, follow one of the desired steps:

Reviewer: I'm not sure about 'desired' here. How about:

```suggestion
To locate the Cashmatic IP address, use one of the following methods:
```

- On the Cashmatic machine, log in with the provided credentials, then go to
  :menuselection:`Settings --> IP address`.
- In the MyCashmatic app or on the website, sign in to your account, then go to
  :menuselection:`Settings --> Network`.

Note the IP address displayed on the machine's or the device's screen, as it is useful for the

> :ref:`configuration in Odoo <pos/cashmatic/odoo_configuration>`.

Reviewer: ```suggestion
Note the IP address displayed on the machine's or the device's screen, as it is needed for the
:ref:`Odoo configuration step <pos/cashmatic/odoo_configuration>`.
```

Just rewording a little, including to align it with Larissa's recent Adyen payment terminals update.

.. _pos/cashmatic/https:

> Connect using HTTPS

Reviewer: Rather than making it sound like this section gives instructions on (how to) 'Connect using HTTPS', maybe just 'Connection over HTTPS'  (and later, 'Connection over HTTP') would be better? (Or am I overthinking it?)

-------------------

To connect the Cashmatic machine using HTTPS, a valid certificate is required. To download the

> certificate, follow the next steps:

Reviewer: ```suggestion
To connect the Cashmatic machine over HTTPS, a valid SSL certificate is required. To generate and download the certificate, follow these steps:
```

Reviewer 2: Suggestion added! But it's not about SSL certificate, it's an HTTPS certificate

> #. Open a web browser and type the IP address in the search bar to access the Cashmatic web

Reviewer: ```suggestion
#. Open a web browser and type the IP address in the address bar to access the Cashmatic web
```

   interface.
#. In the interface's side menu, go to :menuselection:`Settings --> Device`.
#. Scroll down to the :guilabel:`Use Cashmatic CA` option and enable it.
#. Click :guilabel:`Save` and refresh the page. A :guilabel:`Download HTTPS Certificate` link
   appears at the bottom left of the side menu.
#. Click the :guilabel:`Download HTTPS Certificate` link. To install the certificate, contact your

>    integration partner, as the process depends on the computer's operating system.

Reviewer: I think I would put that last line separately, as the instructions are for (generating and) downloading.

```suggestion
#. Click the :guilabel:`Download HTTPS Certificate` link. 

To install the certificate on the device running Odoo Point of Sale, contact your integration partner, as the process depends on the device's operating system.
```

Not 100% sure about my additions about 'device' ... to discuss.
Also, who do we think they mean by 'integration partner'? Someone from Cashmatic?

Reviewer 2: It's a fair addition ;) And usually, when shop owners purchase such machines, they're supported by an integration partner (probably from Cashmatic, or a Cashmatic partner) (a bit like Odoo where the BAs help build the database, connect hardware etc).

.. _pos/cashmatic/http:

> Connect using HTTP

Reviewer: As before

------------------

> To enable HTTP, follow one of the desired steps:

Reviewer: ```suggestion
To connect the Cashmatic machine over HTTP, use one of the following methods:
```

- On the Cashmatic machine, log in, go to :menuselection:`Settings --> Trusted mode`, then click
  :guilabel:`Yes`.
- In the MyCashmatic app or on the website, sign in to your account, go to
  :menuselection:`Settings --> System`, then enable :guilabel:`Unsecured APIs`.

.. note::
   Enable HTTP on the cash machine to allow communication through :doc:`Local Network Access (LNA)

>    <../../hardware_network/pos_lna>`.

Reviewer: ```suggestion
.. note::
   Enabling HTTP on the cash machine allows communication through :doc:`Local Network Access (LNA)
   <../../hardware_network/pos_lna>`.
```

Or, if we add the note at the start, as I suggested, maybe we don't need to repeat this here?

.. seealso::
   `Cashmatic user manual <https://cashmatic.it/wp-content/uploads/2026/02/EN-6.pdf>`_

.. _pos/cashmatic/odoo_configuration:

Odoo configuration
==================

> To connect the Cashmatic machine to Odoo, follow the next steps:

Reviewer: ```suggestion
To connect the Cashmatic machine to Odoo, follow these steps:
```

> #. :ref:`Install <general/install>` the :guilabel:`POS Cashmatic Cash Machines` module.

Reviewer: I many cases in the doc (in general), when we give a module name, we also give its technical name to avoid any confusion. What do you think about adding that here? (And ideally, on the Glory and Cashdro pages too.)

#. Go to :menuselection:`Point of Sale --> Configuration --> Payment Methods` and click
   :guilabel:`New`.
#. Set the :guilabel:`Journal` field to :guilabel:`Cash`.
#. Select the relevant POS in the :guilabel:`Point of Sale` field.
#. Set the :guilabel:`Integration` field to :guilabel:`Cash Machine (Cashmatic)`.

> #. In the :guilabel:`Cashmatic Settings` tab, enter the Cashmatic machine's IP address in the

Reviewer: ```suggestion
#. In the :guilabel:`Cashmatic Settings` tab, enter the Cashmatic machine's :ref:`IP address <pos/cashmatic/ip-address> in the
```

=========
Cashmatic
=========

> **Cashmatic** :doc:`cash machines <../cash_machines>` enable the automation of cash transactions.

Reviewer 2: ```suggestion
`Cashmatic <https://cashmatic.eu/>`_ :doc:`cash machines <../cash_machines>` enable the automation of cash transactions.
```
Let's include a hyperlink to the machine's official website

.. note::
   - The integration with the Cashmatic cash machine only supports payments and refunds.
   - Tasks like filling and emptying the machine must be performed directly through the cash
     machine interface or via the MyCasmatic app/website available on
     `android <https://play.google.com/store/apps/details?id=com.cashmatic.app&pcampaignid=web_share>`_

>      or `ios <https://apps.apple.com/us/app/mycashmatic/id1661294811>`_.

Reviewer 2: ```suggestion
     `Android <https://play.google.com/store/apps/details?id=com.cashmatic.app&pcampaignid=web_share>`_
     or `iOS <https://apps.apple.com/us/app/mycashmatic/id1661294811>`_.
```

.. _pos/cashmatic/configuration:

Cash machine configuration
==========================

.. important::
   - Setting up the Cashmatic machine requires the hardware administrator password and technical
     knowledge. Contact your Cashmatic integration partner for configuration support if assistance is

>      needed.

Reviewer 2: Is there any link we could add here or does it depend on the integration partner?

   - The Cashmatic machine must be assigned a static IP address to ensure stable operation.
   - The Cashmatic machine must be on the same local network as the device running Odoo Point of Sale.

Configuring the Cashmatic machine as a :ref:`payment method in Odoo <pos/cashmatic/odoo_configuration>`
requires locating its IP address. Two connection options are available:

- **HTTPS**: requires a valid SSL certificate.
- **HTTP**: requires enabling HTTP on the cash machine to use
  :doc:`Local Network Access (LNA) <../../hardware_network/pos_lna>`.

To locate the Cashmatic machine's IP address, either check directly on the machine under
:menuselection:`Settings --> IP address`, or log in to the **MyCashmatic** website or app and

> go to :menuselection:`Settings --> Network`.

Reviewer 2: ```suggestion
To locate the IP address on the Cashmatic machine, go to :menuselection:`Settings --> IP address`. To locate the IP address using the **MyCashmatic** app or website, log into your account, and go to :menuselection:`Settings --> Network`.
```

To connect the Cashmatic machine using HTTPS, a valid SSL certificate is required. Two options
are available:

- **Self-signed certificate**: Navigate to ``https://<cashmatic-ip>:50301`` in your browser,
  replacing ``<cashmatic-ip>`` with the machine's IP address, and manually accept the security
  warning to trust the certificate.
- **Certificate issued by Cashmatic**: Contact your Cashmatic integration partner to request the
  generation of a trusted SSL certificate for your machine.

To enable the HTTP setting on the Cashmatic machine, follow the next steps:

#. On the machine, go to :menuselection:`Settings --> Trusted mode` and set it to
   :guilabel:`Yes`.
#. On the **MyCashmatic** app, go to :menuselection:`Settings --> System` and enable

>    :guilabel:`Unsecured APIs`.

Reviewer 2: As discussed on Discord, this part is missing the purpose of the IP address, and why the HTTPS/HTTP connections are options worth considering. It'd be nice to clarify this a bit.

.. _pos/cashmatic/odoo_configuration:

Odoo configuration
==================

To connect the Cashmatic machine to Odoo, follow the next steps:

#. :ref:`Install <general/install>` the :guilabel:`POS Cashmatic Cash Machines` module.
#. Go to :menuselection:`Point of Sale --> Configuration --> Payment Methods` and click
   :guilabel:`New`.
#. Set the :guilabel:`Journal` field to :guilabel:`Cash`.
#. Select the relevant POS in the :guilabel:`Point of Sale` field.
#. Set the :guilabel:`Integration` field to :guilabel:`Cash Machine (Cashmatic)`.
#. In the :guilabel:`Cashmatic Settings` tab, enter the Cashmatic machine's IP address in the
   :guilabel:`Cashmatic IP` field.
#. Enter your Cashmatic user credentials in the :guilabel:`Cashmatic Username` and :guilabel:`Cashmatic

>    Password` fields.

Reviewer 2: Are these the same user credentials used to log into your MyCashmatic account?

> #. Enable :guilabel:`Cashmatic Local Network Access` if your want to connect the machine over `HTTP`.

Reviewer 2: ```suggestion
#. Enable :guilabel:`Cashmatic Local Network Access` to connect the machine over `HTTP`.
```
